slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```
@@ -142,7 +188,8 @@ ::grpc::Status SLAMServiceImpl::GetJpegMap(const GetMapRequest *request,
         }
     } catch (std::exception &e) {
         std::ostringstream oss;
-        oss << "error encoding image " << e.what();
+        oss << "error encoding image: " << e.what();
+        viam::b_continue_session = false;
```

Same here, I would just re-throw the exception.